test/common/stats/thread_local_store_speed_test.cc

```diff
@@ -0,0 +1,88 @@
+// Note: this should be run with --compilation_mode=opt, and would benefit from a
+// quiescent system with disabled cstate power management.
+
+#include "common/common/logger.h"
+#include "common/common/thread.h"
+#include "common/event/dispatcher_impl.h"
+#include "common/stats/heap_stat_data.h"
+#include "common/stats/stats_options_impl.h"
+#include "common/stats/thread_local_store.h"
+#include "common/thread_local/thread_local_impl.h"
+
+#include "test/common/stats/stat_test_utility.h"
+#include "test/test_common/simulated_time_system.h"
+
+#include "testing/base/public/benchmark.h"
+
+namespace Envoy {
+
+class ThreadLocalStorePerf {
+public:
+  ThreadLocalStorePerf() : store_(options_, heap_alloc_) {}
+
+  ~ThreadLocalStorePerf() {
+    store_.shutdownThreading();
+    if (tls_) {
+      tls_->shutdownGlobalThreading();
+    }
+  }
+
+  void accessCounters() {
+    Stats::TestUtil::forEachSampleStat(
+        1000, [this](absl::string_view name) { store_.counter(std::string(name)); });
+  }
+
+  void initThreading() {
+    dispatcher_ = std::make_unique<Event::DispatcherImpl>(time_system_);
+    tls_ = std::make_unique<ThreadLocal::InstanceImpl>();
+    store_.initializeThreading(*dispatcher_, *tls_);
+  }
+
+private:
+  Stats::StatsOptionsImpl options_;
+  Event::SimulatedTimeSystem time_system_;
+  Stats::HeapStatDataAllocator heap_alloc_;
```

Comment:
Is it worth it to have a variant of this test that use shared memory? I'm guessing for most of the things we care about (cached fast path) it doesn't really matter, but possibly worth putting in a TODO like you did for the MT thread contention case below?

Reply:
yes, good idea; done.